src/arm-codegen.c

```diff
@@ -27,18 +27,39 @@ void cfg_flatten()
          ph2_ir = ph2_ir->next) {
         switch (ph2_ir->op) {
         case OP_load_constant:
+            if (ph2_ir->src0 < 0)
+                elf_offset += 12;
+            else if (ph2_ir->src0 > 255)
+                elf_offset += 8;
+            else
+                elf_offset += 4;
+            break;
         case OP_global_address_of:
+            if (ph2_ir->src0 < 0)
+                abort();
+            else if (ph2_ir->src0 > 255)
+                elf_offset += 12;
+            else
+                elf_offset += 4;
+            break;
         case OP_assign:
-        case OP_global_store:
             elf_offset += 4;
             break;
+        case OP_global_store:
+            if (ph2_ir->src1 < 0)
+                abort();
+            else if (ph2_ir->src1 > 4095)
+                elf_offset += 16;
+            else
+                elf_offset += 4;
+            break;
         default:
             printf("Unknown opcode\n");
             abort();
         }
     }
-    /* jump to main */
-    elf_offset += 4;
+    /* prepare argc, argv and jump to main */
```

Comment:
Refine the grammar.

Reply:
They are "prepare argc, argv" and "jump to main". Are they too confusing?

Reply:
Yes, use Grammarly or similar tools to refine.